numpy/core/arrayprint.py

```diff
@@ -805,6 +788,19 @@ def __call__(self, x):
         return r + i
 
 
+class LongComplexFormat(object):
+    def __init__(self, x, precision, suppress_small, sign=False):
+        self.real_format = LongFloatFormat(x.real, precision, suppress_small,
+                                           sign=sign)
+        self.imag_format = LongFloatFormat(x.imag, precision, suppress_small,
+                                           sign='+')
+
+    def __call__(self, x):
+        r = self.real_format(x.real)
+        i = self.imag_format(x.imag)
+        return r + i + 'j'
```

Comment:
Why is this different from the `ComplexFormat` version?

Reply:
I didn't notice they were different. I'll probably copy/subclass the ComplexFloat version.